retrofit-converter-jspoon/src/main/java/pl/droidsonroids/retrofit2/JspoonResponseBodyConverter.java

```diff
@@ -1,19 +1,35 @@
 package pl.droidsonroids.retrofit2;
 
 import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+
+import okhttp3.HttpUrl;
+import okhttp3.MediaType;
 import okhttp3.ResponseBody;
 import pl.droidsonroids.jspoon.HtmlAdapter;
 import retrofit2.Converter;
 
 class JspoonResponseBodyConverter<T> implements Converter<ResponseBody, T> {
-    private HtmlAdapter<T> htmlAdapter;
 
-    JspoonResponseBodyConverter(HtmlAdapter<T> htmlAdapter) {
+    private final HttpUrl httpUrl;
+    private final HtmlAdapter<T> htmlAdapter;
+
+    JspoonResponseBodyConverter(HttpUrl httpUrl, HtmlAdapter<T> htmlAdapter) {
+        this.httpUrl = httpUrl;
         this.htmlAdapter = htmlAdapter;
     }
 
     @Override
     public T convert(ResponseBody responseBody) throws IOException {
-        return htmlAdapter.fromHtml(responseBody.string());
+        Charset charset = null;
+        MediaType mediaType = responseBody.contentType();
+        if (mediaType != null)
+            charset = mediaType.charset();
+
+        try (InputStream is = responseBody.byteStream()) {
```

Comment:
try-with-resources won't work on Android < 4.4 (API level 19) however, at the time of writing market share of older versions is approx. 7% and it is not specified in readme which is minimal required version.
Maybe we should establish it on API 19. What do you think @burnoo ?

Reply:
I think this library shouldn't depend on Android API version and not using try-with-resources is not a big deal to achieve it. So I would opt for changing it to classic try-catch.